repair/repair.cc

```diff
@@ -2434,16 +2434,20 @@ future<> repair_service::repair_tablet(gms::gossip_address_map& addr_map, locato
     auto ranges_parallelism = std::nullopt;
     auto start = std::chrono::steady_clock::now();
     task_metas.push_back(tablet_repair_task_meta{keyspace_name, table_name, table_id, master_shard_id, range, repair_neighbors(nodes, shards), replicas});
-    auto task = co_await _repair_module->make_and_start_task<repair::tablet_repair_task_impl>({}, id, keyspace_name, table_names, streaming::stream_reason::repair, std::move(task_metas), ranges_parallelism);
+    auto task_impl_ptr = seastar::make_shared<repair::tablet_repair_task_impl>(_repair_module, id, keyspace_name, table_names, streaming::stream_reason::repair, std::move(task_metas), ranges_parallelism);
+    auto task = co_await _repair_module->make_task(task_impl_ptr, {});
+    task->start();
```

Comment:
Why doesn't make_and_start_task work?

Reply:
Ah, you need the implementation to get a result.

Reply:
Perhaps make_and_start_task could return a pointer that is templated on the implementation, but that's out of scope here. @Deexie

Reply:
Yes.

Reply:
> Perhaps make_and_start_task could return a pointer that is templated on the implementation, but that's out of scope here. @Deexie

I see no contraindications, opened an issue not to forget about it: https://github.com/scylladb/scylladb/issues/22146.